frontend/src/components/layout/topmessage/CsatSurvey.test.tsx

```diff
@@ -760,4 +761,272 @@ describe("The CSAT survey", () => {
     expect(veryDissatisfiedButton).not.toBeInTheDocument();
     expect(verySatisfiedButton).not.toBeInTheDocument();
   });
+
+  describe("User interactions", () => {
+    beforeEach(() => {
+      const useFirstSeen = // TypeScript can't follow paths in `jest.requireMock`:
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (jest.requireMock("../../../hooks/firstSeen.ts") as any).useFirstSeen;
+      useFirstSeen.mockReturnValue(new Date(0));
+      global.gaEventMock.mockClear();
+    });
```

Comment:
suggestion (non-blocking): this could be a follow-up refactor. this pattern shows up many times (twice here; many times across code-base). we should move this to a test helper function:

```typescript
  function mockFirstSeen(daysAgo: number) {
    const useFirstSeen =
      (jest.requireMock("../../../hooks/firstSeen.ts") as any).useFirstSeen;
    useFirstSeen.mockReturnValueOnce(
      new Date(Date.now() - daysAgo * 24 * 60 * 60 * 1000)
    );
  }
```

and then we can call it like so:

```typescript
  describe("User interactions", () => {
    beforeEach(() => {
      mockFirstSeen(0);
      ...
    });
```